src/common/regions/region_db.ts

```diff
@@ -14,6 +14,11 @@ class RegionDB {
     return region || null;
   }
 
+  findStateByStateCode(stateCode: string): State | null {
+    const region = this.states.find(state => state.stateCode === stateCode);
+    return region || null;
+  }
```

Comment:
If you like my proposal below, we might not need this function yet (but we can keep it)